tests_metricflow/query_rendering/test_metric_filter_rendering.py

```diff
@@ -229,4 +229,28 @@ def test_distinct_values_query_with_metric_filter(
     )
 
 
-# TODO: tests for filters with conversion metrics
+@pytest.mark.sql_engine_snapshot
+def test_metric_filtered_by_itself(
+    request: FixtureRequest,
+    mf_test_configuration: MetricFlowTestConfiguration,
+    dataflow_plan_builder: DataflowPlanBuilder,
+    sql_client: SqlClient,
+    dataflow_to_sql_converter: DataflowToSqlQueryPlanConverter,
+    query_parser: MetricFlowQueryParser,
+) -> None:
+    """Tests a query for a metric that filters by the same metric."""
+    query_spec = query_parser.parse_and_validate_query(
+        metric_names=("bookers",),
+        where_constraint=PydanticWhereFilter(
+            where_sql_template="{{ Metric('bookers', ['guest']) }} > 1.00",
```

Comment:
Just to have a slightly less awkward query (the current one will return 0 rows), shall we do `bookings` instead of `bookers` or something?

It is nice to see the distinct counts applied in the right place, so there is that. Another option is to key on `listing` instead of `guest`.

Reply:
Yeah fair 😂 if you don't mind I'm going to add that on top of the stack, because I spent half of my day rebasing these PRs and I don't wanna do it again 😅

Reply:
I'm a big fan of adding fixup changes to the top of the stack.